src/dmhandler.ts

```diff
@@ -0,0 +1,398 @@
+        //     return;
+        // }
+        // log.info(`Got invite for ${event.room_id} (${event.sender} invited ${event.state_key})`);
+
+        // let discordUserId = null;
+        // try {
+        //     const discordUsers = await this.bridge.getUserStore().getRemoteUsersFromMatrixId(
+        //         event.state_key.substr("@".length),
+        //     );
+        //     if ( discordUsers.length === 0) {
+        //         log.warn("Got an invite for a virtual user the bridge doesn't know about!");
+        //         return;
+        //     }
+        //     discordUserId = discordUsers[0].getId();
+        // } catch (e) {
+        //     log.warn("There was an error trying to fetch a remote user from the store!", e);
+        //     return;
+        // }
+
+        // const result = await this.CheckInvite(event);
+        // if (result.valid === false) {
+        //     const userIntent = this.bridge.getIntent(event.state_key);
+        //     try {
+        //         await userIntent.sendMessage(event.room_id, {
+        //             msgtype: "m.notice",
+        //             body: result.message,
+        //         });
+        //     } catch (e) {
+        //         // Uh..we can't send a message? Oh well, leave anyway.
+        //     }
+        //     return userIntent.leave(event.room_id);
+        // }
+
+        // const existingDMRoom = new DbDmRoom();
+        // await existingDMRoom.RunQuery(this.store, {user_id: event.sender, discord_id: discordUserId});
+        // if (existingDMRoom.Result === true) {
+        //     log.info("User already has a DM room with this user, dropping the old one.");
+        //     const userIntent = this.bridge.getIntent(event.state_key);
+        //     await existingDMRoom.Delete(this.store);
+        //     try {
+        //         await userIntent.kick(existingDMRoom.RoomId, event.state_key, "New DM Room created.");
+        //     } catch (e) {
+        //         log.warn("Failed to kick self from old DM room.", e);
+        //     }
+        // }
+
+        // const newDMRoom = new DbDmRoom();
+        // newDMRoom.RoomId = event.room_id;
+        // await newDMRoom.Insert(this.store);
+    }
+
+    /**
+     * An event that should be forwarded to a viable DMRoom. If a DM room doesn't exist
+     * then one should be created locally and on the discord side.
+     * @param event A matrix event
+     */
+    public async OnMatrixMessage(event): Promise<void> {
+        log.verbose(`Got DM message from ${event.room_id}`);
+        try {
+            const dmRoom = await this.GetDMRoomByRoomId(event);
+            dmRoom.OnMatrixEvent(event);
+        } catch (e) {
+            log.error(`"Failed to get DM room, dropping message! ${event.room_id}`, e);
+        }
+    }
+
+    public GetMatrixIdForUser(user: User): any {
+        // TODO: We should find out what the prefix actually is?
+        return `@_discord_${user.id}:${this.bridge.opts.domain}`;
```

Comment:
unify this with the other matrix id getting so that in case we make `_discord_` modifyable it'll only be one place to edit?